### PR TITLE
feat(j-s): add defence user file access cutoff after dismissal/cancellation

### PR DESCRIPTION
**Pull request by oddsson**
> **Part of a series adding `COMPLETING_FOR_SOME` indictment support:**
> 1. island-is/island.is#22115 — Foundation: types, DTOs, data model ✅
> 2. island-is/island.is#22126 — Backend: partial completion logic & event logs
> 3. **This PR** — Backend: defence user file access cutoff
> 4. Court UI: COMPLETING_FOR_SOME conclusion flow _(upcoming)_
> 5. Defender UI + case table: visibility restrictions _(upcoming)_

## What

Adds a file access cutoff for defence users once all their defendants have been dismissed or cancelled.

- Add `getDefenceUserCutoffDate` utility: returns the latest dismissal/cancellation event date when all of a defender's defendants have been dismissed or cancelled — `undefined` if any defendant is still active
- Apply the cutoff in `canLimitedAccessUserViewCaseFile` so files created after that date are hidden from the defence user
- Pass `fileCreated` through `LimitedAccessViewCaseFileGuard` into the view check
- Apply the same cutoff in `LimitedAccessCaseService` when filtering allowed case files
- Export `getDefenceUserCutoffDate` from the file module
- Add comprehensive tests for all cutoff date edge cases

## Why

Depends on the event log types from island-is/island.is#22115 and the event log writes from island-is/island.is#22126. Once a defendant is dismissed or cancelled, the defence user should no longer see files that were uploaded after that point — this PR enforces that boundary at the file access layer.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

* **New Features**
  * Enforced time-based file access restrictions for limited-access and defence users following dismissal/cancellation events.
  * Case files created after a computed cutoff date tied to a defender are excluded from user access.

* **Tests**
  * Added comprehensive tests covering cutoff date calculation and access behavior across multiple defender and event scenarios.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/file/guards/caseFileCategory.ts [hunk 1]
```diff
@@ -70,6 +72,44 @@ const prisonAdminCaseFileCategories = [
 
 const prisonStaffCaseFileCategories = [CaseFileCategory.APPEAL_RULING]
 
+export const getDefenceUserCutoffDate = (
+  nationalId: string,
+  defendants?: Defendant[],
+): Date | undefined => {
+  const myDefendants = defendants?.filter(
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/file/guards/caseFileCategory.ts:89]: What if the defence user is representing a civil claimant. I think in that case we should show all files. So, perhaps start by checking if any civil claimants match and if so return undefined. Otherwise, do all the stuff below.

**Review by gudjong — APPROVED** [review]
I have one change request, but accepting so you can just fix and continue.

File: apps/judicial-system/backend/src/app/modules/file/guards/caseFileCategory.ts (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/file/guards/caseFileCategory.ts:89]: Good catch! Thanks 🙏
